include/misc/util.h

```diff
@@ -58,7 +58,7 @@ constexpr size_t ARRAY_SIZE(T(&)[N]) { return N; }
 	((ptr) && ((ptr) >= &array[0] && (ptr) < &array[ARRAY_SIZE(array)]))
 
 #define CONTAINER_OF(ptr, type, field) \
-	((type *)(((char *)(ptr)) - offsetof(type, field)))
+	((type *)(((char *)(ptr)) - (offsetof(type, field))))
```

Comment:
What's the logic behind this?

Reply:
So, enclose the *parameters* (whoa, one can contain comma the sequential operator, indeed!). Here, you enclose the "function call". "function call" is highest-precedence operator in C, so it doesn't make sense to enclose it in parens. Of course "function call" may turn out to be a "macro call" instead, and that the reason that even less strict styles than MISRA require macro body to be enclosed in parens. I.e. offsetof() or any other macro would defined as:

~~~
#define offsetof(x, y) (...)
~~~

or considered invalid definition.

Reply:
Of course, "you" is mis-addressed, as sometimes happen, sorry. Keep forgetting to write it in addressee-neutral style. Let me patch the response now.